jit-persistence/src/main/java/org/dcsa/jit/persistence/repository/TimestampDefinitionRepository.java

```diff
@@ -0,0 +1,28 @@
+package org.dcsa.jit.persistence.repository;
+
+import org.dcsa.jit.persistence.entity.TimestampDefinition;
+import org.dcsa.jit.persistence.entity.enums.*;
+import org.dcsa.jit.persistence.entity.enums.PortCallServiceTypeCode;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface TimestampDefinitionRepository
+    extends JpaRepository<TimestampDefinition, String> {
+
+  List<TimestampDefinition>
+      findByEventClassifierCodeAndOperationsEventTypeCodeAndPortCallPhaseTypeCodeAndPortCallServiceTypeCodeAndFacilityTypeCode(
+          String eventClassifierCode,
+          OperationsEventTypeCode operationsEventTypeCode,
+          PortCallPhaseTypeCode portCallPhaseTypeCode,
+          PortCallServiceTypeCode portCallServiceTypeCode,
+          FacilityTypeCode facilityTypeCode);
+
+  @Modifying
+  @Query("INSERT INTO ops_event_timestamp_definition (event_id, timestamp_definition) VALUES (:eventID, :timestampDefinitionID)")
```

Comment:
You probably need to look at nativeQuery since this references a table and not an entity:
https://www.bezkoder.com/jpa-native-query/